app/domain/repos/conjur_ca.rb

```diff
@@ -14,12 +14,28 @@ def self.create(resource_id)
     #
     def self.ca(resource_id)
       ca_info = ::Conjur::CaInfo.new(resource_id)
-      stored_cert = Resource[ca_info.cert_id].last_secret.value
-      stored_key = Resource[ca_info.key_id].last_secret.value
+      stored_cert, stored_key = fetch_ca_secrets(ca_info, resource_id)
       ca_cert = OpenSSL::X509::Certificate.new(stored_cert)
       ca_key = OpenSSL::PKey::RSA.new(stored_key)
       ::Util::OpenSsl::CA.new(ca_cert, ca_key)
     end
 
+    class << self
```

Comment:
Are we sure we want to use this architacutre of class + private? Is it TDD - can we write tests?

Reply:
Changed